learned.lic

```diff
@@ -4,27 +4,105 @@
 
 custom_require.call(%w[drinfomon])
 
-def format_skill_data(skill, time)
-  gain = DRSkill.gained_exp(skill.name)
-  format("%s %s (%0.2f\/hr\, %0.2f\/day)", skill.name.center(18), gain.to_s.center(6), gain / time, gain / time * 24)
-end
+class LearnedData
 
-learning_time = ((Time.now - DRSkill.start_time) / 60.0 / 60)
-columns = get_settings.learned_column_count
+  def initialize
+    @skills_to_show = Array.new
+    
+    setup
+    display_data
+  end
 
-DRSkill.list.select { |item| item.current - item.baseline > 0 }.sort_by { |item| item.current - item.baseline }.reverse.each_slice(columns) do |skills|
-  output = skills.map { |skill| format_skill_data(skill, learning_time) }.join
-  respond(output)
-end
+  def setup
+    arg_definitions = [
+      [
+        { name: 'combats', regex: /(combat|comb)/, optional: true, description: 'Show exp gained in combat skills.' },
```

Comment:
Can you call this with multiple args? E.g. `;learned combats survival`? If I'm reading the `if`/`elsif` block starting on line 38, it would only report one type at a time.

Reply:
It only shows one type at a time, you'd have to rerun it.

Reply:
Are you asking me to implement this or asking if it does already support this?

Reply:
Sorry for the confusion, let me rephrase.

The `arg_definitions` say that you can call it with any number of those optional args (e.g. `;learned combats survival`). But the `if`/`elsif` block prevents that from actually working.

What do you think about changing the `arg_definitions` to this?
```
    arg_definitions = [
      [ { name: 'combats', regex: /(combat|comb)/, description: 'Show exp gained in combat skills.' } ],
      [ { name: 'survival', regex: /(survival|surv)/, description: 'Show exp gained in survival skills.' } ],
      [ { name: 'magic', regex: /magic/, description: 'Show exp gained in magic skills.' } ],
      [ { name: 'armor', regex: /armor/, description: 'Show exp gained in armor skills.' } ],
      [ { name: 'weapons', regex: /(weapons|weap)/, description: 'Show exp gained in weapon skills.' } ],
      [ { name: 'lore', regex: /lore/, description: 'Show exp gained in lore skills.' } ],
      [ { name: 'guild', regex: /guild/, description: 'Show exp gained in your guild skill.'} ],
      []
    ]
```

Reply:
Then it requires at least one?

Reply:
The final `[]` _should_ allow it to run without any args. I can't test that at the moment.